x-pack/plugins/siem/server/lib/hosts/helpers.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { AggregationRequest } from '../types';
+
+export const buildFieldsTermAggregation = (esFields: ReadonlyArray<string>): AggregationRequest =>
+  esFields.reduce<AggregationRequest>(
+    (res, field) => ({
+      ...res,
+      ...getAggregationTypeFromField(field),
+    }),
+    {} as AggregationRequest
```

Comment:
This `as AggregationRequest` cast isn't needed since you used `<AggregationRequest>`. You can just do:

```ts
export const buildFieldsTermAggregation = (esFields: ReadonlyArray<string>): AggregationRequest =>
  esFields.reduce<AggregationRequest>(
    (res, field) => ({
      ...res,
      ...getAggregationTypeFromField(field),
    }),
    {}
  );
```